sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py

```diff
@@ -23,7 +25,16 @@ def _signed_request(self, request):
 
         # Get the path and query from url, which looks like https://host/path/query
         query_url = str(request.http_request.url[len(self._credentials.host) + 8 :])
-
+        # Need URL() to get a correct encoded key value, from "%2A" to "*", when transport is in type AioHttpTransport.
```

Comment:
Is it only for aiohttp?

I guess we should always decode the url?

Reply:
It's for fixing 5 failing async tests when dropping the old transport type - AsyncioRequestsTransport

Reply:
I mean why we only apply this for aiohttp?

Can we do

```py
try:
       query_url = str(URL(query_url))
except : (need to check which error we should catch)
      pass
```
?

Reply:
No, we can't.

We only apply this for aiohttp because only **async** need the correct encoded value `*` after URL(). If we skip the transport check, will do URL() for sync as well, which will cause **sync** tests unauthorized as the expected encoded value is `%2A`.

Reply:
Thank you. Then given here is the only place that uses aiohttp. 

Can we move 

from azure.core.pipeline.transport import AioHttpTransport

here?

Reply:
No. -  We need to check if the transport in request is in type of AioHttpTransport to decide whether need to use URL().

Reply:
I meant 

```py
try:
    from azure.core.pipeline.transport import AioHttpTransport
    if isinstance(request.context.transport, AioHttpTransport) or \
          isinstance(getattr(request.context.transport, "_transport", None), AioHttpTransport) or \
          isinstance(getattr(getattr(request.context.transport, "_transport", None), "_transport", None),
                   AioHttpTransport):
    query_url = str(URL(query_url))
except (ImportError, TypeError):
    pass
```

Reply:
Just tried, all sync tests will fail with aiohttp ImportError.

Reply:
Isn't it caught by except (ImportError, TypeError):?

Reply:
![image](https://user-images.githubusercontent.com/20915761/192372978-1ac24829-3241-47f9-bdae-857189cb28ca.png)

Reply:
I'm thinking if we really need to catch the `ImportError`. - aioHttp is one of the dependencies in dev_requirenemts.txt, so the `ImportError` will not occur if not manually uninstall aioHttp.

Reply:
The issue is that, when just you just `pip install azure-appconfiguration` (like most users would), that doesn't install the requirements from `dev_requirements.txt` -- just those from `setup.py`. We don't want to require `aiohttp` in `setup.py` since users can choose a different transport, so they won't necessarily have `aiohttp` installed after installing the package. Hence, why we need to account for an `ImportError` here

Reply:
Ah, I see. aiohttp is not in setup.py. It's necessary for users then. Thanks